src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseService.java

```diff
@@ -501,15 +501,23 @@ private void setupTestTemplateAndPush(Repository repository, Resource[] resource
             // First get files that are not dependent on the project type
             String templatePath = getProgrammingLanguageTemplatePath(programmingExercise.getProgrammingLanguage()) + "/test";
 
-            String projectTemplatePath = templatePath + "/projectTemplate/**/*.*";
+            // Java both supports Gradle and Maven as a test template
+            String projectTemplatePath = templatePath;
+            ProjectType projectType = programmingExercise.getProjectType();
+            if (projectType != null && projectType.isGradle()) {
+                projectTemplatePath += "/gradle";
+            }
+            else {
+                projectTemplatePath += "/maven";
+            }
+            projectTemplatePath += "/projectTemplate/**/*.*";
             Resource[] projectTemplate = resourceLoaderService.getResources(projectTemplatePath);
-            fileService.copyResources(projectTemplate, prefix, repository.getLocalPath().toAbsolutePath().toString(), false);
+            // keep the folder structure
+            fileService.copyResources(projectTemplate, "projectTemplate", repository.getLocalPath().toAbsolutePath().toString(), true);
```

Comment:
`repository.getLocalPath().toAbsolutePath().toString()` looks wierd for me, is this really necceary / no better way?

Reply:
Why does this look weird for you? `getLocalPath` returns the path of the local repository location on the server instance, `toAbsolutePath` is required to be able to access the files independent of the current working directory.